androidApp/src/main/java/co/nimblehq/avishek/phong/kmmic/android/ui/screen/splash/SplashScreen.kt

```diff
@@ -1,78 +1,260 @@
+            password = password,
+            onEmailChange = { email = it },
+            onPasswordChange = { password = it },
+            onLogInClick = {
+                // TODO: implement in the integrate PR
+            }
+        )
+    }
 }
 
 @Composable
 fun SplashContent(
     shouldShowLogo: Boolean,
     modifier: Modifier = Modifier,
+    animateAlpha: Float,
+    animateLogoScale: Float,
+    animateLogoOffset: Offset,
+    animateBlur: Float,
 ) {
     Box(
         contentAlignment = Alignment.Center,
         modifier = modifier.fillMaxSize()
     ) {
-        Image(
-            painter = painterResource(id = R.drawable.bg_splash),
-            contentDescription = null,
-            contentScale = ContentScale.Crop,
-            modifier = Modifier.matchParentSize()
-        )
-
-        val gradient = Brush.verticalGradient(colors = listOf(Transparent, Black60))
-        Box(modifier = modifier.background(gradient))
-
-        AnimatedVisibility(
-            visible = shouldShowLogo,
-            enter = fadeIn(animationSpec = tween(DURATION_LOGO))
+        Box(
+            contentAlignment = Alignment.Center,
+            modifier = modifier.fillMaxSize()
         ) {
             Image(
-                painter = painterResource(id = R.drawable.logo_white),
+                painter = painterResource(id = R.drawable.bg_splash),
                 contentDescription = null,
-                contentScale = ContentScale.Crop
+                contentScale = ContentScale.Crop,
+                modifier = Modifier
+                    .matchParentSize()
+                    .blur(animateBlur.dp)
             )
+
+            val gradient = Brush.verticalGradient(
+                colors = listOf(
+                    Black.copy(alpha = animateAlpha * InitialTopGradientAlpha),
+                    Black.copy(
+                        alpha = (animateAlpha * BottomGradientAlphaMultiplier) +
+                            InitialBottomGradientAlpha
+                    )
+                )
+            )
+            Box(
+                modifier = Modifier
+                    .matchParentSize()
+                    .background(gradient)
+            )
+
+            AnimatedVisibility(
+                visible = shouldShowLogo,
+                enter = fadeIn(animationSpec = tween(LogoDurationInMillis))
+            ) {
+                Image(
+                    painter = painterResource(id = R.drawable.logo_white),
+                    contentDescription = null,
+                    modifier = Modifier
+                        .offset(animateLogoOffset.x.dp, animateLogoOffset.y.dp)
+                        .scale(animateLogoScale)
+                )
+            }
         }
     }
 }
 
+@Composable
+private fun LoginForm(
+    email: String,
+    password: String,
+    onEmailChange: (String) -> Unit,
+    onPasswordChange: (String) -> Unit,
+    onLogInClick: () -> Unit,
+    modifier: Modifier = Modifier,
+) {
```

Comment:
That way `LoginForm` became the `Stateful Composable` which handle the state of it in itself and expose the required `email` and `password` by `onLogInClick` callback.
```suggestion
private fun LoginForm(
    onLogInClick: (String, String) -> Unit,
    modifier: Modifier = Modifier,
) {
```